tests/ci/devicefarm_job.py

```diff
@@ -0,0 +1,107 @@
+import boto3
+import os
+import requests
+import string
+import random
+import time
+import datetime
+import time
+import json
+from cdk.util.devicefarm_util import AWS_REGION, ANDROID_TEST_NAME, ANDROID_APK, ANDROID_TEST_APK, DEVICEFARM_PROJECT, DEVICEFARM_DEVICE_POOL
+
+
+config = {
+    "awsRegion": AWS_REGION,
+    "namePrefix": ANDROID_TEST_NAME,
+    "appFilePath": ANDROID_APK,
+    "testPackage": ANDROID_TEST_APK,
+    "projectArn": DEVICEFARM_PROJECT,
+    "poolArn": DEVICEFARM_DEVICE_POOL,
+}
+
+client = boto3.client('devicefarm')
+
+unique = config['namePrefix']
+aws_region = config['awsRegion']
+upload_time_limit = datetime.timedelta(minutes=30)
+run_time_limit = datetime.timedelta(hours=1)
+
+print(f"The unique identifier for this run is going to be {unique} -- all uploads will be prefixed with this.")
+
+def upload_df_file(filename, type_, mime='application/octet-stream'):
+    response = client.create_upload(projectArn=config['projectArn'],
+        name = os.path.basename(filename),
+        type=type_,
+        contentType=mime
+        )
+    # Get the upload ARN, which we'll return later.
+    upload_arn = response['upload']['arn']
+    # We're going to extract the URL of the upload and use Requests to upload it
+    upload_url = response['upload']['url']
+    with open(filename, 'rb') as file_stream:
+        print(f"Uploading {filename} to Device Farm as {response['upload']['name']}... ",end='')
+        put_req = requests.put(upload_url, data=file_stream, headers={"content-type":mime})
+        print(' done')
+        if not put_req.ok:
+            raise Exception("Couldn't upload, requests said we're not ok. Requests says: "+put_req.reason)
+    started = datetime.datetime.now()
```

Comment:
Nit: Move this to directly after starting the upload to get more accurate results.